cache/models/samples.js

```diff
@@ -507,6 +507,65 @@ module.exports = {
   },
 
   /**
+   * Delete sample related links
+   * @param  {Object} params - Request parameters
+   * @returns {Promise} - Resolves to a sample object
+   */
+  deleteSampleRelatedLinks(params) {
+    const sampleName = params.key.value;
+    let currSampObj;
+    let aspectObj;
+
+    return redisOps.getHashPromise(sampleType, sampleName)
+    .then((sampObj) => {
+      if (!sampObj) {
+        throw new redisErrors.ResourceNotFoundError({
+          explanation: 'Sample not found.',
+        });
+      }
+
+      currSampObj = sampObj;
+      const aspectName = sampleName.split('|')[ONE];
+      return redisOps.getHashPromise(aspectType, aspectName);
+    })
+    .then((aspObj) => {
+      if (!aspObj) {
```

Comment:
why are we checking for aspect only? do we also need to check whether the subject exists?